src/core/qgsaggregatecalculator.cpp

```diff
@@ -43,7 +43,8 @@ void QgsAggregateCalculator::setParameters( const AggregateParameters &parameter
 
 QVariant QgsAggregateCalculator::calculate( QgsAggregateCalculator::Aggregate aggregate,
     const QString &fieldOrExpression,
-    QgsExpressionContext *context, bool *ok ) const
+    QgsExpressionContext *context, bool *ok,
+    const QgsFeatureIds ids ) const
```

Comment:
Adding a new parameter to make use of the setFilterids of the request, this allows to perform aggregation on only the features contained in a single symbol.